ChangeLog.d/fix-legacy-compression-issue.txt

```diff
@@ -0,0 +1,7 @@
+Bugfix
+   * Fix an issue where ssl_tls13_parse_client_hello() assumed legacy_compression_methods
+     length would always be zero, which is true for TLS 1.3. However, with TLS 1.3 enabled
+     by default, all ClientHello requests (including TLS 1.2 requests) are initially
+     processed by ssl_tls13_parse_client_hello() before being passed to the TLS 1.2
+     parsing function. This caused an issue where legacy_compression_methods
+     might not be zero for TLS 1.2 requests, as it is processed earlier.
```

Comment:
This explains how the code was wrong, but in a way that is meaningless for users. For example, if the name of an internal functions appears in a changelog entry, you're probably doing it wrong.

See https://github.com/Mbed-TLS/mbedtls/issues/9223 for a short write-up of this bug from a user's perspective.

Reply:
Also please indicate the issue number(s) in the changelog entry: “Fixes #9243.” at the end. Or “Fixes #8995, #9243” if we are happy that #8995 is the same issue.

Reply:
OK with “an issues” → “an issue”

Reply:
Oops :)